### PR TITLE
Added trust datastructure

### PR DESCRIPTION
**Pull request by akowal3**
# Summary

TLDR; added trust datastructure and some functions to get useful info out of it. I tried to have everything as similar to the initial trust list, so there wasn't much changing needed.

I was starting to work on other things, but this appears to be the basics. So in client.go when you see something like:
```golang
func (c *client) updateTrustFromSavedHistory() {
	if c.savedHistory.updated {

	}
}
```
be not afraid! There will be more 🤣 

## Test Plan

- Added tests where needed

### CODE REVIEW

**Review by Fabio752 — APPROVED** [review]
I agree it looks better this way and It seems not to break the latest changes I made.

File: internal/clients/team4/client.go [hunk 1]
```diff
@@ -142,20 +155,24 @@ func (c *client) Initialise(serverReadHandle baseclient.ServerReadHandle) {
...
 	}
+
+	trustMatrix := trust{trustMap: trustMap}
+
+	c.trustMatrix = &trustMatrix
+
+	c.idealRulesCachePtr = deepCopyRulesCache(c.ServerReadHandle.GetGameState().RulesInfo.AvailableRules)
+
+	c.updateParents()
+}
```
Comment (tobhil98) [inline internal/clients/team4/client.go]: Reduce whitespace?

File: internal/clients/team4/client.go [hunk 2]
```diff
@@ -46,22 +44,37 @@ func newClientInternal(clientID shared.ClientID, testing *testing.T) client {
...
+	emptyTrust := trust{
+		trustMap: map[shared.ClientID]float64{},
+	}
+
+	importancesMatrix := importances{
+		requestAllocationImportance:                mat.NewVecDense(6, []float64{5.0, 1.0, -1.0, -1.0, 5.0, 1.0}),
+		commonPoolResourceRequestImportance:        mat.NewVecDense(6, []float64{4.0, 1.0, -1.0, -1.0, 1.0, 1.0}),
+		resourceReportImportance:                   mat.NewVecDense(6, []float64{5.0, 5.0, -5.0, -5.0, 1.0, 5.0}),
+		getTaxContributionImportance:               mat.NewVecDense(4, []float64{-2.0, -2.0, 4.0, 1.0}),
+		decideIIGOMonitoringAnnouncementImportance: mat.NewVecDense(3, []float64{1.0, -1.0, 1.0}),
```
Comment (tobhil98) [inline internal/clients/team4/client.go:63]: Could you add some comments to this? It'd be nice to have some reasoning behind the values

**Review by tobhil98 — APPROVED** [review]

File: internal/clients/team4/client.go (hunk 2 above)
Comment (akowal3) [reply internal/clients/team4/client.go:63]: that's to @Fabio752

File: internal/clients/team4/client.go (hunk 1 above)
Comment (akowal3) [reply internal/clients/team4/client.go]: my whitespace? it's set to 4 spaces

File: internal/clients/team4/trust.go [hunk 3]
```diff
@@ -63,3 +63,10 @@ func (t *trust) normalize() {
 		}
 	}
 }
+
+func (t *trust) initialise() {
+	for _, clientID := range shared.TeamIDs {
+		t.trustMap[clientID] = 0.5
+	}
+	t.normalize()
```
Comment (tobhil98) [inline internal/clients/team4/trust.go]: If you spell initialise with s, normalise should also be with s

Comment (akowal3) [reply internal/clients/team4/trust.go]: ok

File: internal/clients/team4/client.go (hunk 2 above)
Comment (tobhil98) [reply internal/clients/team4/client.go:63]: Ah, okay. I couldn't see where the values were deleted from. Looked like a section you'd written

File: internal/clients/team4/client.go (hunk 1 above)
Comment (tobhil98) [reply internal/clients/team4/client.go]: Newline whitespace I was thinking of

**Pull request merged**